### PR TITLE
OS: 12891717 Missing conversion for CopyOnAccess Array on ForInObjectEnumerator

### PR DESCRIPTION
**Pull request by akroshg**
We get copyonaccessnativeintarray when ForInObjectEnumerator. This happens when we are at proxy. Fixed that by converting that array to normal native int array.

### CODE REVIEW

**Comment by akroshg** [comment]
@suwc @kunalspathak please have a look.

**Comment by suwc** [comment]
:shipit:

File: lib/Runtime/Library/JavascriptArray.cpp [hunk 1]
```diff
@@ -12143,6 +12143,10 @@ namespace Js
 
     JavascriptEnumerator * JavascriptArray::GetIndexEnumerator(EnumeratorFlags flags, ScriptContext* requestContext)
     {
+#if ENABLE_COPYONACCESS_ARRAY
+            JavascriptLibrary::CheckAndConvertCopyOnAccessNativeIntArray<Var>(this);
```
Comment (suwc) [inline lib/Runtime/Library/JavascriptArray.cpp]: >      [](start = 7, length = 5)

nit:tab

File: lib/Runtime/Library/JavascriptArray.cpp [hunk 2]
```diff
@@ -12152,6 +12156,9 @@ namespace Js
 
     BOOL JavascriptArray::GetNonIndexEnumerator(JavascriptStaticEnumerator * enumerator, ScriptContext* requestContext)
     {
+#if ENABLE_COPYONACCESS_ARRAY
+        JavascriptLibrary::CheckAndConvertCopyOnAccessNativeIntArray<Var>(this);
```
Comment (kunalspathak) [inline lib/Runtime/Library/JavascriptArray.cpp]: Can we also add a check inside `JavsacriptArray::GetEnumerator()`?

**Pull request merged**